lua/wfp.lua

```diff
@@ -182,6 +182,181 @@ function printArgs(tArgs, tLog)
     print("")
 end
 
+function backup(tArgs, tLog, tWfpControl, tFlasher)
+	--create a backup for all flash areas in netX
+	--read the flash areas and save the images to reinstall them later
+	--Steps:
+		-- read the control file
+		--detect the exisiting flashes
+		-- read the offset and size for each area inside the flash
+		--copy the contents to different bin files
+		--copy xml file
+    
+  
+    local ulSize
+    local ulOffset
+    local DestinationFolder = tArgs.strBackupPath
+    local DestinationXml = DestinationFolder .. "/wfp.xml"
+
+    local fOk = true --be optimistic
+	--overwrite :
+	--check if the directory exists
+	-- if the overwrite parameter is given then delete the directory otherwise throw an error
+    if pl.path.exists(DestinationFolder) == DestinationFolder then
+        if tArgs.fOverwrite ~= true then
+            tLog.error(
+                'The output directory "%s" already exists. Use "--overwrite" to force overwriting it.',
+                DestinationFolder
+            )
+            fOk = false
+        else
+            local tFsResult, strError = pl.dir.rmtree(DestinationFolder)
+            if tFsResult == nil then
+                tLog.error('Failed to delete the output directory "%s": %s', DestinationFolder, strError)
+                fOk = false
+            end
+        end
+    end
+    -- TODO: switch the order of creating a folder and reading the control file -> no need to create a folder if the XML is not readable
+    -- TODO: OR delete DestinationFolder is the result of the backup function is False in the end
```

Comment:
These two considerations are independent of each other.
Do we want the backup to be deleted if anything went wrong in the process?

Reply:
we have already implemented something for that. those TODOSs can be deleted